pkg/kubelet/kubelet_pods.go

```diff
@@ -148,7 +148,16 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 	// - if it is Windows, ContainerD is used.
```

Comment:
This still confuses me, mainly the containerd bit. It sounds like we shouldn't be trying to mount the hosts file on Windows, but I guess if the runtime is containerd we try to? I don't want to end up breaking some scenario folks rely on with this, or we forget to revert this if true host networking for Windows Server containers ever arrives.

Reply:
This confuses me also, I was told that HostNetworking was not supported on containerd at all so I'm really unsure why this tries?

Reply:
am i missing something here @jsturtevant, just found this added this back in for windows containerd https://github.com/kubernetes/kubernetes/pull/49140?

Reply:
just updated to basically revert the windows change here as its a bit cleaner but I don't understand why it was changed back then if hostNetworking was never supported on containerd for windows

Reply:
single file mapping was fixed in the Windows OS and support was added back for Containerd only: https://github.com/kubernetes/kubernetes/pull/83730